### PR TITLE
Beekeeping + Reagent Bees! [WIP]

### PR DESCRIPTION
**Pull request by RemieRichards**
- Bees pollinate nearby plants improving the health of the plant, lowering pests, and increasing yieldmod (Overall boost to plant "quality")
- Bees produce honeycomb which can be grinded to collect honey and the bee's reagent.
- Bees can have a reagent which they will use for attacks and can be collected from their produced honeycombs
- Honey is nutritious and decays into sugar, it's also a VERY mild healing chemical
- Bee reagent honeycombs allow you to farm reagents, including medicines, acids and poisons!
- Bee keeper uniform to prevent bees from hurting you!
- Beeboxes contain honey frames that control how much honeycomb+bees the box can support
- Bees health no longer determines how many appear in the swarm (Sorry Jordie this was laggy as all hell!)
- [X] Bees pollinate plants by "attacking" the hydroponics trays
- [X] Bees get tired and rest in their beebox home
- [X] Bees produce honeycomb
- [X] Honeycomb can be grinded to collect honey and bee reagents
- [X] Bees can have a reagent associated with them
- [ ] Sprites (@WJohn is working on these)
- [ ] Finish Beekeeper uniforms
- [ ] Add a way for users to mutate their bee collections (I was thinking Mutagen+A_Reagent = Bees with A_Reagent) 
- [ ] Add orderable Beeboxes
- [ ] Upgradable honey frames with more capacity?

ART SUBJECT TO CHANGE:
beees_gif

### CODE REVIEW

**Comment by optimumtact** [comment]
Don't bee so hasty

**Comment by SconesC** [comment]
How much reagent will bees inject?

**Comment by optimumtact** [comment]
I have a problem with this in that it'll be fucking ridiculous to try and fight them with any point click weapons, other than that concern :+1:

**Comment by Iamgoofball** [comment]
>reagent bees

I LOVE YOU

**Comment by Fox-McCloud** [comment]
@optimumtact 

Pest spray will probably insta-kill them or something.

**Comment by optimumtact** [comment]
mouse opacity is set so they will take full tile hits

**Comment by Lumipharon** [comment]
Fuck yeah. I'm having flashbacks to my autism setups in minecraft - so many bees.
Most importantly, will you be able to mutate bees - killer hornets or explosive stings come to mind.

**Comment by RemieRichards** [comment]
@Lumipharon once I have the method for mutation down you'll be able to give the bees any reagent you can obtain naturally, the bees use these for stings AND honeycombs produce the reagent.

**Comment by GunHog** [comment]
BEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEES :+1:

**Comment by Killbasa** [comment]
Is there any to prevent them from just ditching botany and ruining someones day?

**Comment by RemieRichards** [comment]
@Killbasa mechanically they view hydro trays with plants as enemies, and they go to "attack" them, so 90% of the time they're much more interested in the plants.
If you're standing next to them (as in, IN BOTANY) without wearing a suit though they will all pretty much turn on you.

**Comment by freerealestate** [comment]
Tonto would be so happy

**Comment by Yoshmaster** [comment]
Beekeeping mode confirmed 2015

**Comment by Xhuis** [comment]
https://www.youtube.com/watch?v=-1GadTfGFvU

**Comment by Boredone** [comment]
So, assuming this is a round start thing, how many suits will Hydro get to defend themselves? Will they be order-able  from cargo?

**Comment by Lumipharon** [comment]
>once I have the method for mutation down you'll be able to give the bees any reagent you can obtain naturally, the bees use these for stings AND honeycombs produce the reagent.

Heuh. Time for botany to defend itself with zombie powder and ClF3 bee swarms.

**Comment by Killbasa** [comment]
Mindbreaker bees sound better to me. And now that i think about it, mute toxin would be pretty funny.

**Comment by RemieRichards** [comment]
@Killbasa the silence of the bees.

**Comment by PKPenguin321** [comment]
dude, this is actually really really cool and i see some huge potential both with nonantag and antag uses
a suggestion though, maybe add a second windoor on the table connecting botany/kitchen so the bees don't get out as easily

**Comment by RemieRichards** [comment]
@PKPenguin321 the bees cannot open the door, and as has been stated previously, they like all hostile mobs pick the closest "target", within botany, this will always be the foolish botanist without a suit OR the plants.

tl;dr they're too busy to fly through the windows. (not to mention in my testing I seem to recall them not having the PASSTABLE flag, meaning they cant cross them anyway, honk!)

**Comment by PKPenguin321** [comment]
sounds good! (but maybe give them passtable so they can wreak even more havoc if released, and because they're flying)

**Comment by Lumipharon** [comment]
That slime reagent that makes slimes super angry should work on bees (or hives), to enrage the bees.

**Comment by xxalpha** [comment]
This is the best feature.

**Comment by SconesC** [comment]
@RemieRichards How much reagent will they inject?

**Comment by PKPenguin321** [comment]
i was thinking about it and i think one issue with this is that it allows easier reagent duplicating, like how the mech syringe gun does but not as finnicky
is there a reagent blacklist for things like adminordrazine and stuff?

**Comment by Jordie0608** [comment]
>Bees health no longer determines how many appear in the swarm (Sorry Jordie this was laggy as all hell!)

RIP 1000 bee mega-death swarm

**Comment by SconesC** [comment]
@PKPenguin321 Honestly I just want to see traitor toxin bees

Initropidril bees...

**Comment by Alek2ander** [comment]
Instead of snowflake bee_friendly maybe use the THICKMATERIAL flag?

**Comment by Jalleo** [comment]
Request can bees take some of the reagent in each of the plants it visit and have a small amount of those reagents be in its honey.
Reason why: https://en.m.wikipedia.org/wiki/Grayanotoxin
Goon has the same sort of idea i believe but this idea does create a lot of possibilities for chaos or medicinal honey

**Comment by Jacquerel** [comment]
Is there any easy way to transport bees, for purposes of traitorously unleashing your napalm bees on an unsuspecting shuttle? Or do you just have to kind of drag the cloud and hope they don't get distracted by those sexy tower caps in the plant box you walk past?

**Comment by bgobandit** [comment]
in real life they have bee transporters, you could code those

**Comment by RemieRichards** [comment]
@SconesC they don't at the moment, they simply call react_mob() on their target.
Not decided if that's final yet.

@PKPenguin321 the initial point of reagent bees was to let you dupe reagents. the production of honeycomb is slow enough and you only get 5u (of the unique reagent, you also get 5u honey) so I say it's fair game. 

As for blacklists, any chem you can obtain in game can be used, whether you've traded TC for badmin chems or bought traitor ones, if it's a stable reagent you can inject/splash with, it can go in a bee.

@Alek2ander good idea, the suit stuff isn't final yet anyway.

@Jalleo people have suggested this, I simply told them "not at this time", I'd like the ability to have bees improving your plants and producing reagents added as the base feature, before I go on to do some of the wackier suggestions.

@Jacquerel 
@bgobandit 
I was just going to let the botanist put the bees in special bee boxes (that break when thrown, mwahaha, bee grenades!) these boxes will also let you move bees from box to box.

**Comment by PKPenguin321** [comment]
so i was thinking about this some more and was wondering about how much of the reagents bees can inject into people
is ODing the station on omnizine with bees feasible

man this feature is gonna be cool

**Comment by WJohn** [comment]
Again remember that right now the sprites you're seeing in that picture are placeholder. I've given remie new pretty sprites to mess with, but they're not in yet.

**Comment by Jalleo** [comment]
I like the bee sprite tho.

**Comment by Bluespace** [comment]
Meth bees when

**Comment by callanrocks** [comment]
Couple ideas, make bee's only aggressive if you attack the hive or something. Give it a chance to trigger a space bear event if lots and lots of bee hives and honey.

**Comment by GunHog** [comment]
So, tell me, how do you weaponize your bees? What would it take for me to be able to turn a section of maintenance into my own personal apiary?

**Comment by Lumipharon** [comment]
If they just attack anyone without a beesuit (in the absence of plants), you basically would just need to spam as loads of hives, and fill them with hilarious chems. Then you have Bee Fort Prime.

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm [hunk 1]
```diff
@@ -436,3 +436,19 @@
...
+
+
+/datum/reagent/consumable/honey
+	name = "Honey"
+	id = "honey"
+	description = "Sweet sweet honey, decays into sugar."
+	color = "#d3a308"
+	nutriment_factor = 15 * REAGENTS_METABOLISM
+
+/datum/reagent/consumable/honey/on_mob_life(var/mob/living/M as mob)
```
Comment (Firecage) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm]: Pleaaaase clean it up.

Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm]: ps. why are you multiplying nutriment factor by the metabolism rate???

Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm]: @Iamgoofball Because nutriment does it too.

**Comment by Supermichael777** [comment]
gunpowder bees ammo for bee gun

**Comment by GunHog** [comment]
This PR is starting to smell like Dismemberment...someone bribe RR to finish!

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm]: that's weird as fuck

**Comment by Razharas** [comment]
@RemieRichards

**Comment by GunHog** [comment]
Yep. Smelling a LOT like dismemberment. Sorry Raz, this one is probably dead now.

**Comment by optimumtact** [comment]
man if only the code was available for people to take over and finish

I think I will create a philosophical movement that lets you take the code and remake it to your own ideal.

I'm gonna call it open source

**Comment by Razharas** [comment]
Ye then create philosophical movement called I Finish What Others Dropped
We dont have enough coder-hobos currently
